src/javascript/app/pages/trade/digit_ticker.js

```diff
@@ -0,0 +1,124 @@
+const DigitTicker = (() => {
+    let $container,
+        $peek,
+        $peek_box,
+        $mask,
+        total_tick_count,
+        contract_status,
+        current_spot;
+
+    const digit_block_size        = 36;
+    const array_of_digits         = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9];
+    const style_offset_correction = 2;
+
+    // Calculate left margin of the peek-box against the "width" of the container
+    const calculateLeftMargin = (width) => (width - (digit_block_size * 10)) / 2;
+
+    // Calculate peek-box left offset.
+    const calculateOffset = () => {
+        const left_offset = calculateLeftMargin($container.offsetWidth);
+        return (+current_spot * digit_block_size) + left_offset - style_offset_correction;
+    };
+
+    const fail = () => {
+        $peek.classList.remove('digit-winning', 'digit-running');
+        $peek_box.classList.remove('digit-winning', 'digit-running');
+        $peek.classList.add('digit-losing');
+        $peek_box.classList.add('digit-losing');
+    };
+
+    const init = (container_id, contract_type, barrier, tick_count, status) => {
+        total_tick_count     = tick_count;
+        current_spot         = '-';
+        contract_status      = status;
+        $container           = document.querySelector(`#${container_id}`);
+        $container.innerHTML = `
+            <div class='peek-box'>
+                <div class='mask'>0/0</div>
+                <div class='peek'></div>
+            </div>
+            <div class='digits'>
+                ${array_of_digits.map(digit => `<div class='digit digit-${digit}'>${digit}</div>`).join('')}
+            </div>
+        `;
+        paintWinningNumbers(winningNumbers(contract_type, barrier));
+        observeResize();
+    };
+
+    const observeResize = () => {
+        window.onresize = () => {
+            calculateOffset();
+        };
+    };
+
+    const paintWinningNumbers = (winning_numbers) => {
+        winning_numbers.forEach(digit => {
+            const element = document.querySelector(`.digit-${digit}`);
```

Comment:
is this something that can be searched within container instead of the whole document?